### ISSUE TITLE
storage::dell::me4::restapi::plugin problem

### ISSUE THREAD

**Issue by Hbitvof** [centreon/centreon-plugins#4406]
Recently we updated our Powervault to firmware version GT280R011-01 and have the same issue as in #3872 . I added the option --digest-sha256 to the command, but this is not solving the problem. Is there a way to debug this further to see what the issue is this time?

**Comment by garnier-quentin** [comment]
Do you still have the issue ?

**Comment by Hbitvof** [comment]
> Do you still have the issue ?

Yes, this issue still exists. It would be nice to get some help solving this problem.

**Comment by garnier-quentin** [comment]
Could you provide the output with --debug ?

**Comment by Hbitvof** [comment]
> Could you provide the output with --debug ?

Here is the output:

UNKNOWN: Login endpoint returns an error: 'Authentication Unsuccessful'
======> request send
GET https://x.x.x.x:443/api/login/64111c362671cbc05dc04c2523011ba1c59ba0a08c180f43b3962e6419109ca1
Accept: application/json
User-Agent: centreon::plugins::backend::http::useragent
Datatype: json

======> response done
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, must-revalidate
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 303
Content-Type: application/json; charset="utf-8"
Client-Date: Thu, 25 May 2023 10:41:57 GMT
Client-Peer: 172.30.69.21:443
Client-Response-Num: 1
Client-SSL-Cert-Issuer: /C=US/ST=CO/O=DELL-EMC/OU=ME4/CN=10.0.0.2
Client-SSL-Cert-Subject: /C=US/ST=CO/O=DELL-EMC/OU=ME4/CN=10.0.0.2
Client-SSL-Cipher: ECDHE-RSA-AES256-GCM-SHA384
Client-SSL-Socket-Class: IO::Socket::SSL
Command-Status: 2 Authentication Unsuccessful
Content-Security-Policy: script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src * 'self' data:; base-uri 'self'; object-src 'self'
Strict-Transport-Security: max-age=31536000;
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1

{

"status":[
  {
    "object-name":"status",
"meta":"/meta/status",
    "response-type":"Error",
    "response-type-numeric":1,
    "response":"Authentication Unsuccessful",
    "return-code":2,
    "component-id":"",
    "time-stamp":"2023-05-25 11:41:57",
    "time-stamp-numeric":1685014917
  }
]
}

**Comment by garnier-quentin** [comment]
It's weird. Could you ask to your dell support about it ?

**Comment by garnier-quentin** [comment]
It should work: https://www.dell.com/support/manuals/fr-fr/powervault-me4024/me4_series_cli_pub/using-a-script-to-access-the-cli?guid=guid-9ae5ccd6-a207-42df-b2f3-1e02a487a354&lang=en-us

Could you try the basic auth to test with curl ?

**Comment by Hbitvof** [comment]
I think I forgot to mention the check is sometimes working. I just did the same command again and now I got response. I use the following command: centreon_plugins.pl --hostname=x.x.x.x --plugin=storage::dell::me4::restapi::plugin --mode=hardware --api-username=username --api-password=password --digest-sha256 --insecure

But when I try it again a few minutes later I receive the Authenticiation Unsuccessful error again.

**Comment by lucie-dubrunfaut** [comment]
Hello :)

Without additional information we are not able to provide a solution or resolve a possible bug.
So @Hbitvof, were you able to resolve the problem with Dell support or do you still need us on this issue?
If it's a Dell issue let me close it here.

**Comment by smigy** [comment]
This issue is still persist, it is weird because sometimes the check is successful but sometimes return a timeout

**Comment by smigy** [comment]
It returns "authentification unsuccessful 
![image](https://github.com/centreon/centreon-plugins/assets/28766849/15941a47-5ba2-4710-8529-e54fec0ea887)
![image](https://github.com/centreon/centreon-plugins/assets/28766849/6b4cd452-e0b0-439b-86d3-6e30afc88cd4)

**Comment by omercier** [comment]
Hi @Hbitvof,
Thank you for your patience.
If you still have this problem, let's have a look at the possible explanations.

The fact thet it sometimes work and sometimes reject the login makes me think of either possibilities:
- The API is unstable. In that case, that is what you could tell to your Dell support.
- Too many requests overloads the device's management card. This should not happen if there are a reasonable number of monitored services taking their data from this device. Can you tell me how many checks are performed on this device by Centreon or other monitoring tools?
- A reasonable number of checks are performed, but each check opens a new session (by requesting the `login` endpoint as it can be seen in the debug you sent). This should not happen since the plugin code stores the session key in a _statefile_ and reuses it until it is not valid any more. Except if something prevents the plugin from storing new values of session keys in the _statefile_. This could happen if the plugin was run manually by another user than `centreon-engine`. You can check this by running this command on the poller that performs the check:

```bash
find /var/lib/centreon/centplugins/ -not -user centreon-engine -ls
```

@Hbitvof @smigy we are quite blind here so do not hesitate to send us the plugin output with the `--debug` option when it works **and** when it doesn't.

**Comment by lucie-tirand** [comment]
Hello :)

Unfortunately, we can’t really make any progress on this issue without further information from users. I’m going to close this issue, but please feel free to reopen it if you can provide us with more details.
